### ISSUE TITLE
feat(flows): conditional Gaussianization flows for probabilistic regression & emulation

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#128]
## Problem / Request

Add conditional Gaussianization flows p(y|x) for probabilistic regression, emulation, and contextual anomaly detection: a high-level constructor plus a matching trainer. The low-level conditioning infrastructure has **already shipped** (`Conditioner`, `ConditionalDiagGaussian`, `ClassCondDiagGaussian`, `ClassCondFlow`, and `SurVAEFlow` threading a single `condition` through base and transforms — see `docs/notebooks/08_conditional_flow_three_ways.ipynb`); this issue is the amortized-MLE *workflow* on top of it.

**Supersedes** #85 (constructor) and #90 (trainer), which were closed as duplicates — their geoscience/anomaly requirements are folded in below.

## User Story

> As a scientist emulating a simulator, doing probabilistic regression, or scoring contextual anomalies, I want one amortized flow trained on (x, y) pairs, so that I can evaluate p(y|x) and sample y|x for held-out x without refitting.

## Motivation

RBIG is unconditional; conditioning + amortization is a genuinely new capability. This issue also unblocks Fair Normalizing Flows (#131), FlowGMM (#132), the NeuTra demo (#133), and the anomaly suite's contextual workflow (#124: score `A(x, c) = −log p(x | c)` instead of `−log p(x)`).

## Proposed API

New file `src/gauss_flows/_src/flows/conditional.py`:

```python
def conditional_gaussianization_flow(
    key, *, n_dims, cond_dim, n_layers=8, n_bins=8, nn_width=64, nn_depth=2,
) -> Transformed

def fit_conditional_flow(
    key, flow, x, condition, *,
    x_val=None, condition_val=None,
    learning_rate=5e-4, max_epochs=100, max_patience=5, batch_size=256,
) -> tuple[Transformed, dict]
```

Usage: `flow.log_prob(y, condition=x)` and `flow.sample(key, condition=x)`.

**Naming decision** (from #85): one constructor, `conditional_gaussianization_flow`, with `cond_dim` — no separate `context_gaussianization_flow` geoscience alias. The docs use "context" language where it helps.

## Mathematical Notes

```text
p(y|x) = p_z(f_x(y)) · |det ∂f_x/∂y|
```

with f_x's coupling/marginal parameters output by a conditioner net g_φ(x).

- Use conditional coupling layers (RQSpline/affine with `cond_dim > 0`) and/or a `ConditionalDiagGaussian` base whose params come from a covariate MLP — both already exist; the constructor composes them.
- **Landmine:** unconditional layers explicitly *drop* `condition` — the constructor must use layers with `cond_shape=(cond_dim,)` so the covariate actually reaches the conditioner MLP.
- **Trainer** (from #90): mirror `fit_gaussianization_flow`'s signature; if `flowjax.train.fit_to_data` doesn't thread conditions directly, wrap the objective or write a light optax loop. Loss: L(θ) = −(1/N) Σᵢ log p_θ(yᵢ | xᵢ).
- Calibration matters: report PIT/coverage; latent recalibration (Dheur & Ben Taieb, arXiv:2505.16636) is a follow-up.

## Anomaly / geoscience workflow (absorbed from #85/#90)

```text
Contextual score:   A(x, c) = −log p_θ(x | c)
Residual variant:   r(x, c) = x − x̂(c);   A(x, c) = −log p_θ(r(x, c) | c)
```

Typical context c: lat/lon, sin/cos(day-of-year), wind, viewing geometry, retrieval uncertainty, or packed `(t, control)` via `pack_time_control`. The context vector is built by #86 (`GeoContextEncoder`); residualization by #83 (`ResidualFlow`); scoring/calibration by #81/#91. This issue only needs to make `conditional_gaussianization_flow` + `fit_conditional_flow` compose cleanly with those.

## References & Existing Code

- Shipped infra: `src/gauss_flows/_src/conditioning.py` (`Conditioner`), `_src/distributions/conditional_diag_gaussian.py`, `_src/flows/survae.py` (condition threading), `_src/flows/class_cond_flow.py`
- Demo: `docs/notebooks/08_conditional_flow_three_ways.ipynb` (base-conditional, coupling-conditional, both)
- Tests: `tests/test_flows_survae_conditional.py`, `tests/test_transforms_coupling_conditional.py`, `tests/test_transforms_conditioner.py`
- Related: #31 (hypernet-based `ConditionalFlowRegressor` recipe — alternative mechanism, same p(y|x) use case; share the heteroscedastic test suite)

## Implementation Steps

- [ ] Add `conditional_gaussianization_flow` in `src/gauss_flows/_src/flows/conditional.py`
- [ ] Add `fit_conditional_flow` (paired `x`/`condition`, validation split, patience, batching)
- [ ] Re-export both via `src/gauss_flows/__init__.py`
- [ ] Extend notebook 08 (or add a sibling) with the amortized-training + calibration workflow

## Acceptance Criteria (numeric)

- [ ] Heteroscedastic 1-D toy (y|x ~ N(sin x, (0.1+|x|)²)): predicted conditional mean/std within 10% of truth; PIT approximately uniform (KS p > 0.05)
- [ ] 90% central predictive-interval empirical coverage in [0.86, 0.94]
- [ ] Amortization: single trained flow evaluates p(y|x) for held-out x without refitting
- [ ] Contextual-anomaly smoke test: on a synthetic with context-dependent normality, conditional score AUROC > unconditional score AUROC
- [ ] `jit`/`vmap` over a batch of conditions

## Definition of Done

- [ ] Code lands at the intended paths
- [ ] Public API exported via `src/gauss_flows/__init__.py`
- [ ] Tests pass: `make test`
- [ ] Lint + typecheck pass: `make lint && make typecheck`
- [ ] Docstrings (Google-style, **Shape:** + **Example:**) on all public symbols

## Testing

- [ ] Unit test: `cond_shape` threading; batched condition
- [ ] Analytic test: Gaussian-linear y|x recovers the closed-form posterior
- [ ] Statistical test: coverage / PIT (mark `slow`)
- [ ] Property test: permutation invariance of independent covariates

## Documentation

- [ ] API reference section
- [ ] Notebook: probabilistic regression / emulation walkthrough (extends notebook 08)

## Out of Scope

Discrete/mixed covariates; recalibration; the context encoder itself (#86); residual baselines (#83, #93).

## Relationships

- Parent (theme epic): #123
- Blocked by: — (infrastructure shipped; see #28, closed completed)
- Blocks: #131, #132, #133; the anomaly epic's conditioning workflow (#124)
- Supersedes: #85, #90 (closed as duplicates)
- Related: #31, #86, #83
